docs/pravega-concepts.md

```diff
@@ -310,12 +267,12 @@ Pravega provides the following ordering guarantees:
 1.  Events with the same Routing Key are consumed in the order they were
     written.
 
-2.  Events with different Routing Keys sent to a specific Reader will always be
+2.  Events with different Routing Keys sent to a specific segment will always be
     seen in the same order even if the Reader backs up and re-reads them.
 
-3.  If there are multiple Readers reading a Stream and they all backup and
-    replay the Stream, they will each see the same Events in the same order each
-    time. 
+3.  If there are multiple Readers reading a Stream and they all backup to any given point, they will never see any reordering with respect to that point. IE: It will never be the case that a event that they read before the chosen point now comes after or vice versa.
+
+4.  The existence and position of an event in a stream is strongly consistent. IE: If an event has been acked to it's writer or has been read by a reader it is guaranteed that it will continue to exist in the same place for all subsequent reads until it is deleted.
```

Comment:
I'm finding the definition of ordering overall a bit confusion. Here is an attempt to do it differently:

- A stream comprises a set of segments that can change over time.
- An event written to a stream is written to a single segment and it is totally ordered with respect to the events of that segment, i.e., for every pair of events e1 and e2, we have that:
   \/ e1 precedes e2 in the segment
   \/ e2 precedes e1 in the segment
   \/ e1 equals e2

- Readers assigned multiple segments interleave the events of the segments while reading, but the order of events per segment respects the one of the segment.